lib/ocpp/v201/smart_charging.cpp

```diff
@@ -193,13 +194,21 @@ SmartChargingHandler::validate_profile_schedules(ChargingProfile& profile,
 }
 
 SetChargingProfileResponse SmartChargingHandler::add_profile(int32_t evse_id, ChargingProfile& profile) {
-    if (STATION_WIDE_ID == evse_id) {
-        station_wide_charging_profiles.push_back(profile);
-    } else {
-        charging_profiles[evse_id].push_back(profile);
-    }
     SetChargingProfileResponse response;
     response.status = ChargingProfileStatusEnum::Accepted;
+
+    auto& profile_storage = STATION_WIDE_ID == evse_id ? station_wide_charging_profiles : charging_profiles[evse_id];
```

Comment:
Man, I really hate the legibility of this ternary operator but I'm not sure of a cleaner way to do it. Does `(STATION_WIDE_ID == evse_id) ? station_wide_charging_profiles : charging_profiles[evse_id]` work?